lxd/instance/instancetype/instance.go

```diff
@@ -1274,15 +1282,8 @@ func ConfigKeyChecker(key string, instanceType Type) (func(value string) error,
 		}
 	}
 
-	if strings.HasPrefix(key, "environment.") {
-		return validate.IsAny, nil
-	}
-
-	if strings.HasPrefix(key, "user.") {
-		return validate.IsAny, nil
-	}
-
-	if strings.HasPrefix(key, "image.") {
+	if (instanceType == Any || instanceType == Container) &&
+		strings.HasPrefix(key, "linux.sysctl.") {
```

Comment:
I think `limits.kernel.` should be treated the same, see https://github.com/canonical/lxd/pull/13051/commits/f0d0128ed636065080607170969244033b2d4616 which I think might have been better done here?

I'm mentioning it here because you introduced `ConfigKeyPrefixesContainer` in the PR.

Reply:
Yeah, `limits.kernel.` is a container only config, good catch.